nectarchain/calibration/container/waveforms.py

```diff
@@ -0,0 +1,350 @@
+        Returns:
+            List[ctapipe_io_nectarcam.NectarCAMEventSource]: List of EventSource for each run files
+        """
+        generic_filename,filenames = DataManagment.findrun(run_number)
+        if run_file is None : 
+            log.info(f"{str(generic_filename)} will be loaded")
+            eventsource = NectarCAMEventSource(input_url=generic_filename,max_events=max_events)
+        else :  
+            log.info(f"{run_file} will be loaded")
+            eventsource = NectarCAMEventSource(input_url=run_file,max_events=max_events)
+        return eventsource
+        
+    def check_events(self):
+        """method to check triggertype of events and counting number of events in all readers
+            it prints the trigger type when trigger type change over events (to not write one line by events)
+
+        Returns:
+            int : number of events
+        """
+        log.info("checking trigger type")
+        has_changed = True
+        previous_trigger = None
+        n_events = 0
+        for i,event in enumerate(self.__reader):
+            if previous_trigger is not None :
+                has_changed = (previous_trigger.value != event.trigger.event_type.value)
+            previous_trigger = event.trigger.event_type
+            if has_changed : 
+                log.info(f"event {i} is {previous_trigger} : {previous_trigger.value} trigger type")
+            n_events+=1
+        return n_events
+
+
+
+    def load_wfs(self,compute_trigger_patern = False):
+        wfs_hg_tmp=np.zeros((self.npixels,self.nsamples),dtype = np.uint16)
+        wfs_lg_tmp=np.zeros((self.npixels,self.nsamples),dtype = np.uint16)
+
+        n_traited_events = 0
+        for i,event in enumerate(self.__reader):
+            if i%100 == 0:
+                log.info(f"reading event number {i}")
+
+            self.event_id[i] = np.uint16(event.index.event_id)
+            self.ucts_timestamp[i]=event.nectarcam.tel[WaveformsContainer.TEL_ID].evt.ucts_timestamp
+            self.event_type[i]=event.trigger.event_type.value
+            self.ucts_busy_counter[i] = event.nectarcam.tel[WaveformsContainer.TEL_ID].evt.ucts_busy_counter
+            self.ucts_event_counter[i] = event.nectarcam.tel[WaveformsContainer.TEL_ID].evt.ucts_event_counter
+
+
+            self.trig_pattern_all[i] = event.nectarcam.tel[WaveformsContainer.TEL_ID].evt.trigger_pattern.T
+
+            for pix in range(self.npixels):
+                wfs_lg_tmp[pix]=event.r0.tel[0].waveform[1,self.pixels_id[pix]]
+                wfs_hg_tmp[pix]=event.r0.tel[0].waveform[0,self.pixels_id[pix]]
+
+            self.wfs_hg[i] = wfs_hg_tmp
+            self.wfs_lg[i] = wfs_lg_tmp
+
+
+
+        #if compute_trigger_patern and np.max(self.trig_pattern) == 0:
+        #    self.compute_trigger_patern()
+
+
+    def write(self,path : str, **kwargs) : 
```

Comment:
Document that this method writes to an output FITS file.

Reply:
Is it enough ?

Reply:
Yes, thanks!